source/mainnet/smart-contracts/tutorials/wCCD/wCCD-interacting.rst

```diff
@@ -0,0 +1,752 @@
+.. _wCCD-interacting:
+
+========================================
+Interacting with the wCCD token protocol
+========================================
+
+.. note::
+
+    Before you start with part 2 of this tutorial, make sure you have access to a testnet node,
+    installed the ``concordium-client``, created an account in the mobile wallet app
+    that is funded with some CCD, and imported the keys
+    into the ``concordium-client`` configuration. If you haven't completed any of the above steps,
+    the piggy bank tutorial :ref:`part 3 <piggy-bank-preparing>`
+    will guide you through these setup steps.
+    If you haven't completed any of the above steps, you can continue
+    with :ref:`part 3 <wCCD-front-end-set-up>` of this tutorial also we recommend completing every part of this tutorial.
+
+Non-state-mutative functions
+----------------------------
+
+The protocol has four non-state-mutative functions (``balanceOf``, ``operatorOf``, ``tokenMetadata``, and ``supports``)
+that you can invoke on the ``proxy`` contract or the ``implementation`` contract.
```

Comment:
I added this comment:
`
It is recommended to invoke the non-state-mutative functions on the ``proxy`` and not on the ``implementation`` in production. When the protocol is upgraded, the ``implementation`` address becomes invalid which means you would need to update your production product to the new ``implementation`` address.`

I feel for the tutorial, it will be easier for most people to invoke the functions on the implementation (because the schema is embedded) but for any production product, it should be recommended to use the proxy or have some mechanism (be aware) to update the product to the new implementation address.

Reply:
This paragraph is a bit unclear to me. Perhaps the following would work better
```
While for testing it can be convenient to invoke these functions on the ``implementation`` contract, you should always invoke them on the ``proxy`` contract in production. When the protocol is upgraded, the ``implementation`` address becomes invalid, you would need to update your production product to the new ``implementation`` address.
```

Reply:
changed to above sentence